src/Umbraco.Core/HealthCheck/Checks/Permissions/FolderAndFilePermissionsCheck.cs

```diff
@@ -32,22 +32,16 @@ public FolderAndFilePermissionsCheck(ILocalizedTextService textService, IOptions
         /// <summary>
         /// Get the status for this health check
         /// </summary>
-        /// <returns></returns>
-        public override IEnumerable<HealthCheckStatus> GetStatus()
-        {
-            //return the statuses
-            return new[] { CheckFolderPermissions(), CheckFilePermissions() };
-        }
+        // TODO: This should really just run the IFilePermissionHelper.RunFilePermissionTestSuite and then we'd have a
+        // IFilePermissions interface resolved as a collection within the IFilePermissionHelper that runs checks against all
+        // IFilePermissions registered. Then there's no hard coding things done here and the checks here will be consistent
+        // with the checks run in IFilePermissionHelper.RunFilePermissionTestSuite which occurs on install too.
+        public override IEnumerable<HealthCheckStatus> GetStatus() => new[] { CheckFolderPermissions(), CheckFilePermissions() };
 
         /// <summary>
         /// Executes the action and returns it's status
         /// </summary>
-        /// <param name="action"></param>
-        /// <returns></returns>
-        public override HealthCheckStatus ExecuteAction(HealthCheckAction action)
-        {
-            throw new InvalidOperationException("FolderAndFilePermissionsCheck has no executable actions");
-        }
+        public override HealthCheckStatus ExecuteAction(HealthCheckAction action) => throw new InvalidOperationException("FolderAndFilePermissionsCheck has no executable actions");
```

Comment:
`NotSupportedException`. 

Or maybe an interface redesign if there are implementations that cannot support the features.

Reply:
Yep the plan is to change healthchecks, there will be not 'action' to execute in netcore. @bergmania has allocated tasks for this